src/services/pallets/PalletsOnGoingReferendaService.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017-2024 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import { BlockHash } from '@polkadot/types/interfaces';
+
+import { IPalletOnGoingReferenda, IReferendaInfo } from '../../types/responses';
+import { AbstractService } from '../AbstractService';
+
+export class PalletsOnGoingReferendaService extends AbstractService {
+	/**
+	 * Fetch all on-going referenda that have track: root (0) and whitelisted (1), along
+	 * with their associated information.
+	 *
+	 * @param hash `BlockHash` to make call at
+	 */
+	async derivePalletOnGoingReferenda(hash: BlockHash): Promise<IPalletOnGoingReferenda> {
+		const { api } = this;
+		const historicApi = await api.at(hash);
+		const { number } = await api.rpc.chain.getHeader(hash);
```

Comment:
Is it worth merging these two promises into a single promise.all for performance's sake? Otherwise all looking good to me